onnxruntime/contrib_ops/rocm/bert/attention_impl.h

```diff
@@ -181,6 +182,33 @@ class CompatRocblasMathModeSetter {
   }
 };
 
+enum AttentionMode {
+  // Q,K,V,PastK,PastV,PresentK,PresentV
```

Comment:
PastK and PastV is symmetric, may classify PastK+PastV as: NONE, 2BNPH, 2BNMH, BNPH, BNMH. 

The only difference of 2BNPH and BNPH is whether the buffers of pastk and pastv is continuous. If your algorithm has not used the continuous property. You can split 2BNPH to BNPH + BNPH (by assigning pointer of pastV. In this way, there are only 3 classes: None, BNPH, BNMH.

Reply:
For this mode, it is just for human convenience. Becasue for the algorithm, all you need to do is set the correct shape, stride and the buffer pointer. This mode is just a helper for developer (me) to figure out the strides and pointer easier.
See `ConvertToOffsetedBufferViews` and `GetQkvStrides`